students/scott/session08/circle.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+"""
+circle stuff
+"""
+
+from math import pi
+
+
+class Circle:
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        return 2 * self.radius
+    
+    @diameter.setter
+    def diameter(self, val):
+        self.radius = val / 2
+
+    @property
+    def area(self):
+        return pi * self.radius**2
+
+    def __str__(self): #'string' method
+        return  "Circle with radius: {}".format(self.radius)
+
+    def __repr__(self): #'represent' method
+        return "Circle({})".format(self.radius)
+
+
+    def __add__(self, another):  #add two circles together by adding the radius of one to the radius of the other
+        sum_radius = self.radius + another.radius
+        return Circle(sum_radius)
+
+    def __mul__(self, mult):
+        new_radius = self.radius*mult
+        return Circle(new_radius)
+
+    def __equal__(self, another):
```

Comment:
These are spelled:


__le__(self, other)
Defines behavior for the less-than-or-equal-to operator, <=.
__ge__(self, other)
Defines behavior for the greater-than-or-equal-to operator, >=.

you need to use those exact names for them to be used.

Reply:
gotcha...didn't know I needed those exact names...

I figured something was up since my linter was colorizing those names differently from other ones....

thanks!

comparing two lists of circles after sorting isn't working....   my circles.sort() method returns None, and I don't know why.

Reply:
List.sort() sorts the list in place.

You can use new_list = sorted(old_list) to get a new one.

Reply:
got it working, thanks!

what's the difference in doing:
new_llst = circles.sort()
vs.
new_list = sorted(circles)

will .sort() only sort it in place, even if I'm trying to assign that sorted list to a new list?